src/components/message/message-card/MessageCard.tsx

```diff
@@ -110,44 +119,55 @@ const MessageCard = ({ message, viewType, setViewType }: Props) => {
 		bookmarksStore.toggleMessagePin(message);
 	}, [bookmarksStore.toggleMessagePin]);
 
-	const addMessagesToExport = React.useCallback(
-		() => messagesStore.exportStore.addMessageToExport(message),
-		[messagesStore.exportStore.addMessageToExport],
-	);
-
 	const isExported = messagesStore.exportStore.isExported(message);
 
-	const rootClass = createBemBlock(
-		'message-card-wrapper',
-		isAttached ? 'attached' : null,
-		isBookmarked ? 'pinned' : null,
-		isHighlighted ? 'highlighted' : null,
-		isSoftFiltered ? 'soft-filtered' : null,
-		messagesStore.exportStore.isExport ? 'export-mode' : null,
-		isExported ? 'exported' : null,
-	);
-
 	return (
-		<div className={rootClass} onClick={addMessagesToExport}>
+		<div className='messages-list__item'>
 			<MessageCardBase
 				message={message}
-				viewType={viewType}
-				setViewType={setViewType}
+				viewTypes={viewTypes}
+				getSavedViewType={getSavedViewType}
 				hoverMessage={hoverMessage}
 				unhoverMessage={unhoverMessage}
+				addMessagesToExport={addMessagesToExport}
+				isHighlighted={isHighlighted}
+				isSoftFiltered={isSoftFiltered}
+				isExported={isExported}
+				isExport={messagesStore.exportStore.isExport}
+				isExpanded={isExpanded}
 				isBookmarked={isBookmarked}
 				isAttached={isAttached}
 				toogleMessagePin={toogleMessagePin}
 				sortOrderItems={sortOrderItems}
 			/>
+			<MessageExpandButton
+				message={message}
+				isExpanded={isExpanded}
+				setExpanded={setExpandedMessage}
+			/>
 		</div>
 	);
-};
+});
 
 const RecoverableMessageCard = (props: OwnProps) => {
-	const viewTypesStore = useMessagesViewTypesStore();
-	const { viewType, setViewType } = viewTypesStore.getSavedViewType(props.message);
-	return <MessageCard {...props} viewType={viewType} setViewType={setViewType} />;
+	const viewStore = useMessagesViewStore();
+	const { getSavedViewType, expandedMessages, setExpandedMessage } = viewStore;
```

Comment:
i think viewStore should only handle viewType change, not isExpandedStatus, we probably need to use `<StateSaver />` here